src/utils/twitch-api.ts

```diff
@@ -0,0 +1,34 @@
+export type ChannelsResponse = {
+  data: Array<ChannelInfo>
+}
+
+export type ChannelInfo = {
+  broadcaster_id: string
+  broadcaster_login: string
+  broadcaster_name: string
+  broadcaster_language: string
+  game_id: string
+  game_name: string
+  title: string
+  delay: number
+  tags: string[]
+}
+
+export async function fetchCurrentChannelInfo(channelId: string, auth: Twitch.ext.Authorized) {
+  const params = new URLSearchParams()
+  params.set('broadcaster_id', channelId)
+
+  const response = await fetch(`https://api.twitch.tv/helix/channels?${params}`, {
+    headers: {
+      'client-id': auth.clientId,
+      'Authorization': `Extension ${auth.helixToken}`,
```

Comment:
How does this behave if the user isn't signed in? Do you get a token that works anonymously?

Reply:
This would most likely not work without a user sessopn. The callback does guarantee a valid `helixToken`, but would probably never be called without a twitch session. I could not test that out correctly using the twitch local extension preview. I will ask the twitch dev discord about it. That would also mean, we do not have the channel id. So it would not break break, but chat commands would not work for anons.

So we would need a fallback if no session is available. Not sure if it would be best to fall back to maya + alveussanctuary in that case or only alveussanctuary? Or we do use the streamer config view to let the alveus team configure it for each channel.